packages/ingest/lock.js

```diff
@@ -12,17 +12,50 @@ async function delay(t) {
   });
 }
 
+/**
+* Count Lock
+* Counts the number of locks in a bucket
+*
+* @param {Object} bucket - The AWS S3 bucket to check
+* @param {String} pName - The provider name
+* @returns {Integer} - Number of current locks in the bucket
+**/
 async function countLock(bucket, pName) {
-  const list = await aws.S3.list(bucket, `${lockPrefix}/${pName}`);
-  return list.Contents.length;
+  var list = await aws.S3.list(bucket, `${lockPrefix}/${pName}`);
+  var count = checkOldLocks(bucket, list.Contents);
+  return count;
+}
+
+/**
+* Check Old Locks
+* Checks all locks and removes those older than five minutes
+*
+* @param {Object} bucket - The AWS S3 bucket with the locks to check
+* @param {String} list - The list of locks in the bucket
+* @returns {Boolean} - Number of locks remaining in bucket
+**/
+async function checkOldLocks(bucket, list) {
+  var count = list.length;
+  var item;
+  for (item in list) {
+    var date = list[item].LastModified;
+    var diff = new Date() - date;
+    const fiveMinutes = 300000; // 5 * 60 seconds * 1000 milliseconds
+    if (diff > fiveMinutes) {
+      aws.S3.delete(bucket, list[item].Key);
+      count--;
+    }
+  }
+  return count;
 }
 
 async function addLock(bucket, pName, filename) {
   return aws.S3.put(bucket, `${lockPrefix}/${pName}/${filename}`, '');
 }
 
 async function removeLock(bucket, pName, filename) {
-  return aws.S3.delete(bucket, `${lockPrefix}/${pName}/${filename}`);
+  console.log("removee lock bu not really");
+  //return aws.S3.delete(bucket, `${lockPrefix}/${pName}/${filename}`);
```

Comment:
I think we still need this line, don't we? otherwise the locks are not removed for successful downloads

Reply:
ooops debugging, one sec